src/app/api/collections/utils.py

```diff
@@ -0,0 +1,19 @@
+from typing import Generic, TypeVar
+
+from elasticsearch_dsl.response import Response
+from glom import glom
+
+from app.api.collections.models import CollectionNode, MissingMaterials
+
+T = TypeVar("T", CollectionNode, MissingMaterials)
+
+
+def map_elastic_response_to_model(
+    response: Response, specs: dict, model: Generic[T]
```

Comment:
not sure if `Generic[T]` is the right type annotation here. eventually it is `Type[T]`. Does pycharm complain when the function is called in `tree.py` ?

Reply:
No, it does not. But it does complain with Type[T]. I generally check that IDE, tests and using the actual endpoints does not raise an exception and works as I believe it should.

Reply:
oki :)